core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
> I will try to another route and remote the Environment.getSettings() method, which means all of this goes away. I will do it in another PR

OK. I looked at this more carefully and this will be a good but big change (still a good one though). I rather not embark on it now. 

This brings me back to the question:

> Can you make a suggestion as to what to add?